src/main/java/mate/academy/springboot/datajpa/repository/specification/requestparam/GetAllProductByCategory.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.springboot.datajpa.repository.specification.requestparam;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.repository.specification.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Repository;
+
+@Repository
```

Comment:
why repository?

Reply:
but why also not?
thin edge for awareness of difference
https://www.baeldung.com/spring-component-annotation
amended